scripts/ci/build_ninja.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+set -ex
+
+ROOTDIR="$(dirname "$0")/../.."
+cd "${ROOTDIR}"
+
+# shellcheck disable=SC2166
+if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" -o -n "$FORCE_RELEASE" ]
+then
+    echo -n "" >prerelease.txt
+else
+    # Use last commit date rather than build date to avoid ending up with builds for
+    # different platforms having different version strings (and therefore producing different bytecode)
+    # if the CI is triggered just before midnight.
+    TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="ci.%cd" >prerelease.txt
+fi
+
+if [ -n "$CIRCLE_SHA1" ]
+then
+    echo -n "$CIRCLE_SHA1" >commit_hash.txt
+fi
+
+mkdir -p build
+cd build
+
+# shellcheck disable=SC2086
+cmake --version
+cmake .. -DCMAKE_BUILD_TYPE="${CMAKE_BUILD_TYPE:-Release}" $CMAKE_OPTIONS -G "Ninja"
+ninja -j 4
```

Comment:
Does ninja read `MAKEFLAGS`? Or maybe it has its own variable for that?

We set this flag in `config.yml` and I think it's a better place - there we can set it appropriately because we know the number of cores corresponding to the `resource_class` or executor we choose.

Reply:
No. CMake will generate ninja files (`ninja.build`) instead of Makefile (default generator).

Reply:
Yeah, but I was hoping ninja might support this (or more likely) its own variable.

Looks like you're right though. The maintainers of ninja are weirdly opposed to any hint of having an env variable, even suggesting odd workarounds like replacing the binary with a custom script in system dirs (https://github.com/ninja-build/ninja/issues/797, https://github.com/ninja-build/ninja/issues/1370).

Anyway, in that case I think this should be passed via a parameter (or our custom variable) from CI config.